## LGTM — ready for human review & squash-merge

Both blockers from the prior review are now fixed. Good persistence on this one.

### What this PR does

Replaces the inline `DROP TABLE IF EXISTS history` in `Memory.reset()` and `AsyncMemory.reset()` with a delegation to `SQLiteManager.reset()`, which atomically drops and recreates both `history` and `messages` tables inside a transaction. Also restores the entity-store reset block in `AsyncMemory.reset()` that was accidentally dropped in the first commit.

### Strengths

- `mem0/memory/main.py:1931` (sync) and `main.py:3473` (async) — correct delegation to `SQLiteManager.reset()`, the right abstraction boundary. If a future migration adds a third table, there's now one place to update.
- `main.py:3481–3486` — entity-store parity fully restored; async callers no longer leave entity-store data behind after reset.
- `tests/test_memory.py:121–144` — the new `test_memory_reset_clears_messages_table` goes through real `Memory.reset()` with a real SQLiteManager on a `tmp_path` db (mocked LLM/vector/embedder only). It asserts both `messages` and `history` are empty. Red-green confirmed: the test fails on the pre-fix code (messages table not cleared) and passes with the fix.

### Issues

Minor (optional):

- `tests/memory/test_storage.py:286–299` — `test_reset_clears_messages_table` inserts into both `history` and `messages`, calls `mgr.reset()`, but only asserts `COUNT(*) FROM messages == 0`. The docstring says "both history and messages tables" but skips the `history` assertion. The gap is fully covered by the integration test above, but the unit test is internally inconsistent. One-liner fix: add `assert mgr.connection.execute("SELECT COUNT(*) FROM history").fetchone()[0] == 0` after line 299.

### Evidence

- **CI:** Python SDK 3.10 / 3.11 / 3.12 PASS. `changelog_check` fails — this is a rebase artifact: `main` bumped to `2.0.7` and added the changelog entry after this branch diverged; the PR itself doesn't touch `pyproject.toml`. A rebase onto current `main` clears it before merge.
- **Targeted tests:** `pytest tests/test_memory.py::test_memory_reset_clears_messages_table` → 1 passed (0.61s); `pytest tests/memory/test_storage.py::TestSQLiteManager::test_reset_clears_messages_table` → 1 passed (0.31s)
- **Red-green proof:** Reverted `mem0/memory/main.py` to `origin/main` (buggy code), test failed with `AssertionError: messages table must be empty after Memory.reset() — assert 1 == 0`. Restored PR code → PASS. ✓
- **Entity store (async):** Block present at `main.py:3481–3486`, structurally matches the sync equivalent at `main.py:1941–1946`.
- **Lint:** `ruff check mem0/memory/main.py tests/test_memory.py tests/memory/test_storage.py` → all checks passed.
- **Could NOT verify:** Live Neo4j/Memgraph behavior for entity-store reset path — code-traced only.

### Assessment

Ready to merge: **Yes** (squash) · Est. human time: ~3 min

Please rebase onto `main` first (one command: `git rebase origin/main`) to clear the `changelog_check` CI failure. The minor unit-test assertion gap can be fixed in this PR or as a follow-up — your call.

--- a/mem0/memory/main.py
+++ b/mem0/memory/main.py
@@ -1923,10 +1923,8 @@ class Memory(MemoryBase):
         """
         logger.warning("Resetting all memories")
 
-        if hasattr(self.db, "connection") and self.db.connection:
-            self.db.connection.execute("DROP TABLE IF EXISTS history")
-            self.db.connection.close()
-
+        self.db.reset()
+        self.db.close()
         self.db = SQLiteManager(self.config.history_db_path)
 
         if hasattr(self.vector_store, "reset"):
@@ -3465,10 +3463,8 @@ class AsyncMemory(MemoryBase):
         if hasattr(self.vector_store, "client") and hasattr(self.vector_store.client, "close"):
             await asyncio.to_thread(self.vector_store.client.close)
 
-        if hasattr(self.db, "connection") and self.db.connection:
-            await asyncio.to_thread(lambda: self.db.connection.execute("DROP TABLE IF EXISTS history"))
-            await asyncio.to_thread(self.db.connection.close)
-
+        await asyncio.to_thread(self.db.reset)
+        await asyncio.to_thread(self.db.close)
         self.db = SQLiteManager(self.config.history_db_path)
 
         self.vector_store = VectorStoreFactory.create(

--- a/tests/memory/test_storage.py
+++ b/tests/memory/test_storage.py
@@ -280,3 +280,13 @@ class TestSQLiteManager:
         assert history[0]["actor_id"] is None
         assert history[0]["is_deleted"] is False
         mgr.close()
+
+    def test_reset_clears_messages_table(self, temp_db_path):
+        """Regression: reset() must drop both history and messages tables."""
+        mgr = SQLiteManager(temp_db_path)
+        mgr.add_history(memory_id="m1", old_memory=None, new_memory="new", event="ADD")
+        mgr.save_messages([{"role": "user", "content": "hello", "name": None}], "sess1")
+        mgr.reset()
+        count = mgr.connection.execute("SELECT COUNT(*) FROM messages").fetchone()[0]
+        assert count == 0, "messages table should be empty after reset"
+        mgr.close()

--- a/tests/test_memory.py
+++ b/tests/test_memory.py
@@ -121,6 +121,30 @@ def test_collection_name_preserved_after_reset(mock_sqlite, mock_llm_factory, mo
 @patch('mem0.utils.factory.EmbedderFactory.create')
 @patch('mem0.utils.factory.VectorStoreFactory.create')
 @patch('mem0.utils.factory.LlmFactory.create')
+def test_memory_reset_clears_messages_table(mock_llm_factory, mock_vector_factory, mock_embedder_factory, tmp_path):
+    """Regression: Memory.reset() must clear the messages table, not just history."""
+    mock_embedder_factory.return_value = MagicMock()
+    mock_vector_factory.return_value = MagicMock()
+    mock_llm_factory.return_value = MagicMock()
+
+    config = MemoryConfig()
+    config.history_db_path = str(tmp_path / "test.db")
+    memory = Memory(config)
+
+    memory.db.save_messages([{"role": "user", "content": "hello", "name": None}], "sess1")
+    memory.db.add_history(memory_id="m1", old_memory=None, new_memory="x", event="ADD")
+
+    memory.reset()
+
+    msg_count = memory.db.connection.execute("SELECT COUNT(*) FROM messages").fetchone()[0]
+    hist_count = memory.db.connection.execute("SELECT COUNT(*) FROM history").fetchone()[0]
+    assert msg_count == 0, "messages table must be empty after Memory.reset()"
+    assert hist_count == 0, "history table must be empty after Memory.reset()"
+
+
+@patch('mem0.utils.factory.EmbedderFactory.create')
+@patch('mem0.utils.factory.VectorStoreFactory.create')
+@patch('mem0.utils.factory.LlmFactory.create')
 @patch('mem0.memory.storage.SQLiteManager')
 def test_search_handles_incomplete_payloads(mock_sqlite, mock_llm_factory, mock_vector_factory, mock_embedder_factory):
     """Test that search operations handle memory objects with missing 'data' key gracefully."""

